components/scoreHandler.ts

```diff
@@ -261,9 +264,36 @@ export async function missionEnd(
             ? 0
             : sessionDetails.npcKills.size + sessionDetails.crowdNpcKills
 
-    // const allMissionStories = getConfig("MissionStories")
-    // const missionStories = (contractData.Metadata.Opportunities || []).map((missionStoryId) => allMissionStories[missionStoryId])
-
+    const locations = getVersionedConfig<PeacockLocationsData>(
+        "LocationsData",
+        req.gameVersion,
+        true,
+    )
+    const location = contractData.Metadata.Location
+    const parent = locations.children[location].Properties.ParentLocation
+    const locationChallenges =
+        controller.challengeService.getGroupedChallengeLists({
+            type: ChallengeFilterType.ParentLocation,
+            locationParentId: parent,
+        })
+    const contractChallenges =
+        controller.challengeService.getChallengesForContract(
+            sessionDetails.contractId,
+            req.gameVersion,
+        )
+    const challengeCompletion =
+        controller.challengeService.countTotalNCompletedChallenges(
+            locationChallenges,
+            userData.Id,
+            req.gameVersion,
+        )
+    const opportunities = contractData.Metadata.Opportunities
+    const opportunityCount = opportunities ? opportunities.length : 0
+    const opportunityCompleted = opportunities
+        ? opportunities.filter(function (ms) {
+              return ms in userData.Extensions.opportunityprogression
+          }).length
```

Comment:
Can probably be an arrow function to improve performance by a few cpu cycles 😆 
```suggestion
        ? opportunities.filter((ms) => (
              ms in userData.Extensions.opportunityprogression
          )).length
```